File: courtlistener/api/schools.py
"""
Schools API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class SchoolsAPI(BaseAPI):
    """API for accessing educational institutions data."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "schools/"
    
    def list(self, **kwargs) -> Dict[str, Any]:
        """
        List schools.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            API response with paginated list of schools
        """
        return self.client.get(self.endpoint, params=kwargs)
    
    def get(self, school_id: int) -> Dict[str, Any]:
        """
        Get a specific school by ID.
        
        Args:
            school_id: School ID
        
        Returns:
            School data
        """
        return self.client.get(f"{self.endpoint}{school_id}/")
    
    def paginate(self, **kwargs):
        """
        Get paginated schools.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            PageIterator for schools
        """
        return self.client.paginate(self.endpoint, params=kwargs)

File: php/src/CourtListener/Api/Schools.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Schools API endpoint
 */
class Schools extends BaseApi
{
    protected string $endpoint = 'schools/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List schools with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listSchools(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific schools by ID
     *
     * @param int|string $id Schools ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getSchools(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search schools
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchSchools(array $params = []) {
        return $this->search($params);
    }
}
